low-level/files/opt/seagate/sspl/setup/sspl_prepare.py

```diff
@@ -47,61 +48,40 @@ def validate(self):
         3. Validate storage controller connectivity
         4. Validate network interface availability
         """
-        machine_id = Utility.get_machine_id()
+        req_conf = {}
         mgmt_interfaces = []
         data_private_interfaces = []
         data_public_interfaces = []
 
         # Validate input/provisioner configs
-        node_type = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-            "server_node>%s>type" % machine_id)
-        cluster_id = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-            "server_node>%s>cluster_id" % machine_id)
-        if node_type.lower() not in ["virtual", "vm"]:
-            bmc_ip = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "server_node>%s>bmc>ip" % machine_id)
-            bmc_user = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "server_node>%s>bmc>user" % machine_id)
-            secret = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
-                "server_node>%s>bmc>secret" % machine_id)
```

Comment:
`Utility.get_config_value` method validates and throw if value is None or key is not found in config.
Where did you move the validation? At each mini provisioner stage, its required to validate the input configs.
```
[root@ssc-vm-2217 ~]# python3
Python 3.6.8 (default, Nov 16 2020, 16:55:22)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>> from cortx.utils.conf_store import Conf
>>> Conf.load("conf", "yaml:///etc/sspl.conf")
>>> Conf.get("conf", "server_node>dummy")
>>>
```
Conf won't throw exception on invalid key or null value for a mandatory key.

Reply:
Agree. 
the setup scripts should not apply these changes anyway, only service files will read config from conf_utils.py.
Reverted the changes and keeping the prepare & setup scripts as original.